stable_baselines3/common/callbacks.py

```diff
@@ -435,3 +435,47 @@ def _on_step(self) -> bool:
             self.last_time_trigger = self.num_timesteps
             return self._on_event()
         return True
+
+
+class StopTrainingOnMaxEpisodes(BaseCallback):
+    """
+    Stop the training once a maximum number of episodes are played.
+
+    For multiple environments presumes that, the desired behavior is that the agent trains on each env for ``max_episodes``
+    and in total for ``max_episodes * n_envs`` episodes.
+
+    :param max_episodes: (int) Maximum number of episodes to stop training.
+    :param verbose: (int)
+    """
+
+    def __init__(self, max_episodes: int, verbose: int = 0):
+        super(StopTrainingOnMaxEpisodes, self).__init__(verbose=verbose)
+        self.max_episodes = max_episodes
+        self._max_episodes_per_env = max_episodes
+        self.episodes_played = 0
+        self._first_step = True
+
+    def _on_step(self) -> bool:
+        # Checking for both 'done' and 'dones' keywords because:
+        # Some models use keyword 'done' (e.g.,: SAC, TD3, DQN, DDPG)
+        # While some models use keyword 'dones' (e.g.,: A2C, PPO)
+        done_array = np.array(self.locals.get("done") if self.locals.get("done") is not None else self.locals.get("dones"))
+        self.episodes_played += np.sum(done_array).item()
+
+        # Sets max episodes per env in first step to avoid doing this operation multiple times
+        if self._first_step:
+            self._max_episodes_per_env = self.max_episodes * done_array.size
+            self._first_step = False
+
+        continue_training = bool(self.episodes_played < self._max_episodes_per_env)
```

Comment:
No need for `bool`, can do just `self.episodes_played < self._max_episodes_per_env`.

Reply:
For code consistency, followed other code examples that made explicit type declaration even for booleans. E.g,.: In `StopTrainingOnRewardThreshold` we have `continue_training = bool(self.parent.best_mean_reward < self.reward_threshold)`.

Reply:
this was needed because it returned a `np.bool` otherwise. Here, I think it is not needed.